Tools/cases_generator/test_generator.py

```diff
@@ -0,0 +1,310 @@
+# Sorry for using pytest, these tests are mostly just for me.
+# Use pytest -vv for best results.
+
+import tempfile
+
+import generate_cases
+
+
+def run_test(input: str, expected: str):
+    temp_input = tempfile.NamedTemporaryFile("w+")
+    temp_input.write(generate_cases.BEGIN_MARKER)
+    temp_input.write(input)
+    temp_input.write(generate_cases.END_MARKER)
+    temp_input.flush()
+    temp_output = tempfile.NamedTemporaryFile("w+")
+    a = generate_cases.Analyzer(temp_input.name, temp_output.name)
+    a.parse()
+    a.analyze()
+    if a.errors:
+        raise RuntimeError("Found {a.errors} errors")
```

Comment:
```suggestion
        raise RuntimeError(f"Found {a.errors} errors")
```

Maybe its time to make f strings default ;)